src/amber/cli/commands/routes.cr

```diff
@@ -43,7 +43,16 @@ module Amber::CLI
       end
 
       private def parse_routes
-        File.read_lines("config/routes.cr").each do |line|
+        # parse each routes file in modules
+        Dir.glob(["src/modules/**/*_routes.cr"]) do |file|
+          parse_routes_file(file)
+        end
+        # read global routes last
+        parse_routes_file("config/routes.cr")
```

Comment:
I would have expected to read the global routes first because which are defined first have a higher priority in the router, and the `config/routes.cr` file seems like a place of higher priority.

Reply:
I purposefully put the module routes first as they will/should be more specific and thus should have a higher priority.  In the case where you use the scaffold generator to create a new module, the routes will all have the module name prefix and it will be up to the developer to ensure the prefix does not clash with existing routes in config/routes.cr.  In the case in the future where a module is added to the application via a plugin then I expect the routes in the plugin module should be namespaced to avoid clashes with other routes.

Reply:
In all situations where there is a prefix and there is no conflict, it's fine.

As a developer with an existing application in production, when I install a plugin which conflicts with my existing routing, I expect my routing to take precedence, not the plugin. 

Moreover, when a conflict inevitably happens, I want a way to override it. When a plugin is installed which conflicts with my existing routing, I shouldn't have to edit the plugin code or generated code to get my existing routes back.

I think our disagreement on this serves to illustrate why implicitly merging routes files is going to cause confusion, regardless of the order of operations.

To amend my previous example for a shard-based plugin or recipe:

```crystal
  routes :web do
    # ...
    include_routes "routes/admin"
    include_routes "../lib/plugin/routes"
    # ...
  end
```

Reply:
not a disagreement.... more a discussion :smile: 
You are right a plugin routing should not conflict with your existing routing.  With plugins I envisage the routes would be namespaced - if we curate the plugins - at least curate a list of plugins that conform to a format -  we can ensure that plugin routes are namespaced.  We can ask that plugin authors namespace the plugin routes with a namespace unique to the plugin.

Reply:
BTW good idea about ../lib/plugin/routes if plugins are installed as shards (but not sure if they are going to be added to the app/shard.yml yet)